sdk/java-sdk/src/main/scala/kalix/javasdk/impl/action/ActionEffectImpl.scala

```diff
@@ -68,6 +65,13 @@ object ActionEffectImpl {
       copy(internalSideEffects = sideEffects)
   }
 
+  final case class EmptyEffect[T](internalSideEffects: Seq[SideEffect]) extends PrimaryEffect[T] {
+    def isEmpty: Boolean = true
+    protected def withSideEffects(sideEffect: Seq[SideEffect]): EmptyEffect[T] =
+      copy(internalSideEffects = sideEffect)
```

Comment:
Do we really allow side effects to empty? I guess we can't easily protect against it with types, so shouldn it lead to an IllegalState runtime exception rather?

Reply:
I thought why not allow side effects, like an alert? But I'm fine with adding an exception or even trying to redefine the types. How about making `PrimaryEffect` extend `EmptyEffect`? And being EmptyEffect having only `def isEmpty`?

Reply:
Forget what I said. Never mind. I'll add the exception.